File: UI/src/app/components/post/post.ts
import { Category } from 'src/app/models/category';
import { Comment } from 'src/app/models/comment';

export class Post {
    constructor(
        public id:number,
        public title:string,
        public content:string,
        public pubDate:Date,
        public lastModified:Date,
        public categoryID:number,
        public comments:Comment[],
    ){}


}

File: BloggingSystem/Models/Post.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace BloggingSystem.Models
{
    public class Post
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]

        [Key]
        public int ID { get; set; } 

        [Required]
        public string Title { get; set; }


        [Required]
        public string Content { get; set; }

        public DateTime PubDate { get; set; } = DateTime.UtcNow;

        public DateTime LastModified { get; set; } = DateTime.UtcNow;

        public Category Category { get; set; }

        public ICollection<Comment> Comments { get; set; }
        public int CategoryID { get;  set; }
    }
}
